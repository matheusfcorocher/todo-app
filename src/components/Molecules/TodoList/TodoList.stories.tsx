import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import TodoList from './TodoList';

export default {
  title: 'Molecules/TodoList',
  component: TodoList,
  decorators: [
    (Story) => (
      <ul style={{ listStyleType: 'none', margin: 0, padding: 0 }}>
        <Story />
      </ul>
    ),
  ]
} as ComponentMeta<typeof TodoList>;

const Template: ComponentStory<typeof TodoList> = (args) => <TodoList {...args} />;

export const Default = Template.bind({});
Default.args = {};

Default.args = {
  todosData: []
};

export const withItems = Template.bind({});
withItems.args = {
  todosData: [
    {
      id: "1",
      title: "TASK",
      isCompleted: false,
    },
    {
      id: "2",
      title: "TASK 2",
      isCompleted: true,
    }
  ]
};