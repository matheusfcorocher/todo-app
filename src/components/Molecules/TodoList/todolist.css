body {
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
    monospace;
}
@import '../../../foundations/colors/colors.css';

.todo-panel {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding: 0.5rem;
  box-shadow: var(--dark-color) 0px 0px 0px 1px inset;
}

.todo-menu {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
}

.complete-button {
  margin-right: 1rem;
}

.title-input {
  display: flex;
  align-items: center;
  height: 100%;
  width: 100%;
  padding: 0;
  border: none;
  font-family: 'Nunito Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif;
  font-size: 1rem;
  font-weight: 400;
}

.todo-list {
  display: flex;
  flex-direction: column;
  gap: 0.5rem;
  margin-top: 0.5rem;
  padding: 0;
  width: auto;
}