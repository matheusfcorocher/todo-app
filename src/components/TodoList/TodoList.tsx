import React, { useState } from 'react';
import { TodoItem } from '../TodoItem/TodoItem';
import '../../App.css';

type TodoData = {
  id: string;
  title: string;
  isCompleted: boolean;
}

function TodoList() {
  const [todos, setTodos] = useState([]);

  const handleCreateTodoItem = (TodoTitle: string) => {

  };

  const listToDoItems = todos.map((todo: TodoData) => 
    <TodoItem 
      task={{
        id: todo.id,
        title: todo.title,
        isCompleted: todo.isCompleted
      }} 
      handleUpdateTodoItemTitle={function (newTitle: string): void {
        throw new Error('Function not implemented.');
      }} 
    
      handleUpdateTodoItemState={function (newState: boolean): void {
        throw new Error('Function not implemented.');
      }} 
      handleDeleteTodoItem={function (newState: string): void {
        throw new Error('Function not implemented.');
      }}    
    />
  );

  return (
    <div className="TodoList">
      <input 
        type="text"
        role="textbox"
        placeholder='Enter todo name'
        value={''}
        name="title"
        className={`title-input`}
        onChange={() => handleCreateTodoItem}
      />
      <ul>
        {listToDoItems}
      </ul>
    </div>
  );
}

export default TodoList;
