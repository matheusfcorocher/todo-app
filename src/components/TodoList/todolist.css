body {
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
    monospace;
}

.todo-list {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  max-width: 30rem;
  padding: 0.5rem;
  box-shadow: #00000026 0px 0px 0px 1px inset;
}

.todo-panel {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
}

.title-input {
  display: flex;
  align-items: center;
  height: 100%;
  width: 100%;
  padding: 0;
  border: none;
  font-family: 'Nunito Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif;
  font-size: 1rem;
  font-weight: 400;
}