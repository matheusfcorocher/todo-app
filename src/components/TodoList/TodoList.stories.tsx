import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import TodoList from './TodoList';

export default {
  title: 'Example/TodoList',
  component: TodoList,
  decorators: [
    (Story) => (
      <ul style={{ listStyleType: 'none', margin: 0, padding: 0}}>
        <Story />
      </ul>
    ),
  ]
} as ComponentMeta<typeof TodoList>;

const Template: ComponentStory<typeof TodoList> = (args) => <TodoList/>;

export const Default = Template.bind({});
Default.args = {};