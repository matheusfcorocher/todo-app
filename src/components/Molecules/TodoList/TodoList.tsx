import React, { useState } from 'react';
import { TodoItem } from '../TodoItem/TodoItem';
import '../../../App.css';
import './todolist.css';

type TodoData = {
  id: string;
  title: string;
  isCompleted: boolean;
}

interface TodoListProps {
  todosData?: Array<TodoData>;
}

function TodoList({ todosData = [] }: TodoListProps) {
  const listToDoItems = todosData.map((todo: TodoData) => {
    return (
      <TodoItem
        key={todo.id}
        task={{
          id: todo.id,
          title: todo.title,
          isCompleted: todo.isCompleted
        }}
        handleUpdateTodoItemTitle={function (newTitle: string): void {
          throw new Error('Function not implemented.');
        }}

        handleUpdateTodoItemState={function (newState: boolean): void {
          throw new Error('Function not implemented.');
        }}
        handleDeleteTodoItem={function (newState: string): void {
          throw new Error('Function not implemented.');
        }}
      />
    );
  }
  );

  return (
    <ul role="list" className={"todo-list"}>
      {listToDoItems}
    </ul>
  );
}

export default TodoList;
