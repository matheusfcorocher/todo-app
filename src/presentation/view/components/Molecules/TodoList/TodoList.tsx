import React, { useState } from "react";
import {
  filterTodoTasksByIsCompleted,
} from "../../../../../domain/entities/TodoTask";
import { IsCompletedFilterControllerReturnType } from "../../../../controllers/IsCompletedFilterController";
import { TodoTaskControllerReturnType } from "../../../../controllers/TodoTaskController";
import { TodoItem } from "../TodoItem/TodoItem";
import "./todolist.css";

interface TodoListProps {
  isCompletedFilterController: IsCompletedFilterControllerReturnType;
  todoTaskController: TodoTaskControllerReturnType;
}

function TodoList({ isCompletedFilterController, todoTaskController }: TodoListProps) {
  const { getTodoTasks } = todoTaskController;
  const todoTasks = getTodoTasks();
  const { getIsCompletedFilter } = isCompletedFilterController;
  const filter = getIsCompletedFilter();
  
  const listToDoItems = filterTodoTasksByIsCompleted({
    todoTasks,
    isCompleted: filter,
  }).map((todo) => {
    return (
      <TodoItem
        key={todo.id}
        task={{
          id: todo.id,
          title: todo.title,
          isCompleted: todo.isCompleted,
        }}
        todoTaskController={todoTaskController}
      />
    );
  });

  return (
    <ul role="list" className={"todo-list"}>
      {listToDoItems}
    </ul>
  );
}

export default TodoList;
