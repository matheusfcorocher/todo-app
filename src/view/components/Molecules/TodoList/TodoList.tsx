import React, { useState } from 'react';
import { TodoItem } from '../TodoItem/TodoItem';
import './todolist.css';
import { HandleDeleteTodo, HandleUpdateTodoState, HandleUpdateTodoTitle } from '../../../../App';
import { TodoTask, TodoTasks } from '../../../../domain/entities/TodoTask';

interface TodoListProps {
  todoTasks?: TodoTasks;
  handleDeleteTodo: HandleDeleteTodo;
  handleUpdateTodoTitle: HandleUpdateTodoTitle;
  handleUpdateTodoState: HandleUpdateTodoState;
}

function TodoList({ todoTasks = [], handleDeleteTodo, handleUpdateTodoTitle, handleUpdateTodoState }: TodoListProps) {
  const listToDoItems = todoTasks.map((todo: TodoTask) => {
    return (
      <TodoItem
        key={todo.id}
        task={{
          id: todo.id,
          title: todo.title,
          isCompleted: todo.isCompleted
        }}
        handleUpdateTodoItemTitle={handleUpdateTodoTitle}
        handleUpdateTodoItemState={handleUpdateTodoState}
        handleDeleteTodoItem={handleDeleteTodo}
      />
    );
  }
  );

  return (
    <ul role="list" className={"todo-list"}>
      {listToDoItems}
    </ul>
  );
}

export default TodoList;
