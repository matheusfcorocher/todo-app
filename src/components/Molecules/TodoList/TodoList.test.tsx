import React from 'react';
import { render, screen } from '@testing-library/react';
import TodoList from './TodoList';

describe("::Components ::Molecules ::TodoList", () => {
  test('renders TodoList', () => {
      render(<TodoList />);
      const todoList = screen.getByRole("list");
      expect(todoList).toMatchSnapshot();
  })
})