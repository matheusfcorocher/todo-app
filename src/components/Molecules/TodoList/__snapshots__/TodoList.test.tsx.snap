// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`::Components ::Molecules ::TodoList renders TodoList 1`] = `
<ul
  class="todo-list"
  role="list"
/>
`;
